src/combinations.rs

```diff
@@ -94,6 +94,56 @@ impl<I: Iterator> Combinations<I> {
         let n = pool.count();
         (n, remaining_for(n, first, &indices).unwrap())
     }
+
+    /// Initialises the iterator by filling a buffer with elements from the
+    /// iterator, return a boolean indicating whether or not we've run out of
+    /// combinations.
+    fn init(&mut self) -> bool {
+        self.pool.prefill(self.k());
+        let done = self.k() > self.n();
+        if !done {
+            self.first = false;
+        }
+
+        done
+    }
+
+    /// Increments indices representing the combination to advance to the next
+    /// (in lexicographic order by increasing sequence) combination. For example
+    /// if we have n=3 & k=2 then [0, 1] -> [0, 2] -> [0, 3] -> [1, 2] -> ...
```

Comment:
If n=3 then the indices are 0 1 2 and there is no 3. So n=4 here.

Reply:
Ah yes, excuse me. Will patch